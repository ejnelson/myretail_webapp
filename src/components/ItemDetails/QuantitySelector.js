import React, { Component } from 'react';
import './ItemDetails.css';

export default class QuantitySelector extends Component {
  constructor(props) {
    super(props);
    this.state = { quantity: 1 };
  }
  increment = () =>
    this.setState(prevState => ({ quantity: prevState.quantity + 1 }));
  decrement = () => {
    if (this.state.quantity > 1) {
      this.setState(prevState => ({ quantity: prevState.quantity - 1 }));
    }
  };

  render() {
    return (
      <div className="QuantitySelector">
        quantity:
        <div className="plusMinus">
          <i
            className="el el-minus-sign decrement"
            onClick={() => this.decrement()}
          />
          <p className="quantity">{this.state.quantity}</p>
          <i
            className="el el-plus-sign increment"
            onClick={() => this.increment()}
          />
        </div>
      </div>
    );
  }
}
