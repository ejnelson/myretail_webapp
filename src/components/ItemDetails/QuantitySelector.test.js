import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import QuantitySelector from './QuantitySelector';

describe('QuantitySelector', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<QuantitySelector />, div);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<QuantitySelector />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('increments and decrements correctly', () => {
    // Render the QuantitySelector
    const selector = shallow(<QuantitySelector />);

    selector.find('.increment').simulate('click');
    expect(selector.text()).toEqual('quantity:2');
    selector.find('.decrement').simulate('click');
    expect(selector.text()).toEqual('quantity:1');
    selector.find('.decrement').simulate('click');
    expect(selector.text()).toEqual('quantity:1');
  });
});
