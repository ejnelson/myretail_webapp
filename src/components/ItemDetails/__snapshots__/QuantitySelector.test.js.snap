// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantitySelector renders correctly 1`] = `
<div
  className="QuantitySelector"
>
  quantity:
  <div
    className="plusMinus"
  >
    <i
      className="el el-minus-sign decrement"
      onClick={[Function]}
    />
    <p
      className="quantity"
    >
      1
    </p>
    <i
      className="el el-plus-sign increment"
      onClick={[Function]}
    />
  </div>
</div>
`;
